MultiSource/Benchmarks/Olden/mst/hash.h

```diff
@@ -1,20 +1,26 @@
 /* For copyright information, see olden_v1.0/COPYRIGHT */
 
+#include <stdchecked.h>
 #include "stdio.h"
+#include <stdio_checked.h>
 
-typedef struct hash_entry {
+struct hash_entry {
   unsigned int key;
   void *entry;
-  struct hash_entry *next;
-} *HashEntry;
+  ptr<struct hash_entry> next;
+};
 
-typedef struct hash {
-  HashEntry *array;
-  int (*mapfunc)(unsigned int);
+typedef ptr<struct hash_entry> HashEntry;
+
+struct hash {
+  HashEntry *array : itype(array_ptr<HashEntry>);
```

Comment:
I'm going to go with the `array_ptr<HashEntry> array : count(size);` type, and then we can fix codegen/inference later without having to update. This will cause a clang exception if anyone attempts to compile/run this benchmark though, which isn't great.